Intellij 2019.2 and 2019:3 NoClassDefFoundError: IconLoader

When trying to compile thrift module:
```
Error:Internal error: (java.util.concurrent.ExecutionException) com.intellij.util.xmlb.XmlSerializationException: Cannot deserialize class com.intellij.plugins.thrift.config.ThriftCompilerOptions
java.util.concurrent.ExecutionException: com.intellij.util.xmlb.XmlSerializationException: Cannot deserialize class com.intellij.plugins.thrift.config.ThriftCompilerOptions
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.jetbrains.jps.model.serialization.JpsProjectLoader.loadModules(JpsProjectLoader.java:401)
	at org.jetbrains.jps.model.serialization.JpsProjectLoader.loadModules(JpsProjectLoader.java:330)
	at org.jetbrains.jps.model.serialization.JpsProjectLoader.loadFromDirectory(JpsProjectLoader.java:198)
	at org.jetbrains.jps.model.serialization.JpsProjectLoader.loadProject(JpsProjectLoader.java:98)
	at org.jetbrains.jps.model.serialization.impl.JpsSerializationManagerImpl.loadModel(JpsSerializationManagerImpl.java:42)
	at org.jetbrains.jps.cmdline.JpsModelLoaderImpl.loadModel(JpsModelLoaderImpl.java:50)
	at org.jetbrains.jps.cmdline.BuildRunner.load(BuildRunner.java:80)
	at org.jetbrains.jps.cmdline.BuildSession.runBuild(BuildSession.java:265)
	at org.jetbrains.jps.cmdline.BuildSession.run(BuildSession.java:121)
	at org.jetbrains.jps.cmdline.BuildMain$MyMessageHandler.lambda$channelRead0$0(BuildMain.java:228)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: com.intellij.util.xmlb.XmlSerializationException: Cannot deserialize class com.intellij.plugins.thrift.config.ThriftCompilerOptions
	at com.intellij.util.xmlb.XmlSerializer.deserialize(XmlSerializer.java:62)
	at com.intellij.plugins.thrift.jps.ThriftFacetConfigurationSerializer.loadExtension(ThriftFacetConfigurationSerializer.java:26)
	at com.intellij.plugins.thrift.jps.ThriftFacetConfigurationSerializer.loadExtension(ThriftFacetConfigurationSerializer.java:16)
	at org.jetbrains.jps.model.serialization.facet.JpsFacetConfigurationSerializer.loadExtension(JpsFacetConfigurationSerializer.java:47)
	at org.jetbrains.jps.model.serialization.facet.JpsFacetSerializer.addExtension(JpsFacetSerializer.java:72)
	at org.jetbrains.jps.model.serialization.facet.JpsFacetSerializer.addFacets(JpsFacetSerializer.java:63)
	at org.jetbrains.jps.model.serialization.facet.JpsFacetSerializer.loadFacets(JpsFacetSerializer.java:44)
	at org.jetbrains.jps.model.serialization.JpsProjectLoader.loadModule(JpsProjectLoader.java:460)
	at org.jetbrains.jps.model.serialization.JpsProjectLoader.lambda$loadModules$2(JpsProjectLoader.java:397)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	... 3 more
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:116)
	at com.intellij.util.ReflectionUtil.newInstance(ReflectionUtil.java:408)
	at com.intellij.util.xmlb.BeanBinding.newInstance(BeanBinding.java:135)
	at com.intellij.util.xmlb.BeanBinding.deserialize(BeanBinding.java:128)
	at com.intellij.util.xmlb.NotNullDeserializeBinding.deserializeUnsafe(NotNullDeserializeBinding.java:14)
	at com.intellij.util.xmlb.AbstractCollectionBinding.deserializeItem(AbstractCollectionBinding.java:226)
	at com.intellij.util.xmlb.CollectionBinding.doDeserializeList(CollectionBinding.java:34)
	at com.intellij.util.xmlb.AbstractCollectionBinding.deserializeList(AbstractCollectionBinding.java:169)
	at com.intellij.util.xmlb.Binding.deserializeList(Binding.java:29)
	at com.intellij.util.xmlb.OptionTagBinding.deserialize(OptionTagBinding.java:97)
	at com.intellij.util.xmlb.BasePrimitiveBinding.deserializeUnsafe(BasePrimitiveBinding.java:60)
	at com.intellij.util.xmlb.BeanBinding.deserializeInto(BeanBinding.java:227)
	at com.intellij.util.xmlb.BeanBinding.deserializeInto(BeanBinding.java:181)
	at com.intellij.util.xmlb.BeanBinding.deserialize(BeanBinding.java:129)
	at com.intellij.util.xmlb.XmlSerializer.deserialize(XmlSerializer.java:56)
	... 17 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at com.intellij.util.ReflectionUtil.newInstance(ReflectionUtil.java:389)
	... 30 more
Caused by: java.lang.NoClassDefFoundError: com/intellij/openapi/util/IconLoader$LazyIcon
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
	at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:515)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:423)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:417)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:691)
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:416)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at com.intellij.plugins.thrift.config.target.GeneratorType.<init>(GeneratorType.java:114)
	at com.intellij.plugins.thrift.config.target.GeneratorType.<init>(GeneratorType.java:12)
	at com.intellij.plugins.thrift.config.target.GeneratorType$1.<init>(GeneratorType.java:13)
	at com.intellij.plugins.thrift.config.target.GeneratorType.<clinit>(GeneratorType.java:13)
	at com.intellij.plugins.thrift.config.target.Java.<init>(Java.java:59)
	... 36 more
Caused by: java.lang.ClassNotFoundException: com.intellij.openapi.util.IconLoader$LazyIcon
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:436)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 51 more
```

--- a/plugin-jps/src/com/intellij/plugins/thrift/config/target/GeneratorType.java
+++ b/plugin-jps/src/com/intellij/plugins/thrift/config/target/GeneratorType.java
@@ -106,24 +106,19 @@ public enum GeneratorType {
   XSD("xsd", "/fileTypes/xsdFile.png"),
   //---
   ;
-  private final Icon myIcon;
+  private final String iconName;
   protected final String name;
 
   GeneratorType(String name, final String iconName) {
     this.name = name;
-    myIcon = new IconLoader.LazyIcon() {
-      @Override
-      protected Icon compute() {
-        return IconLoader.getIcon(iconName);
-      }
-    };
+    this.iconName = iconName;
   }
 
   public Generator create() {
     return new Generator(this);
   }
 
-  public Icon getIcon() {
-    return myIcon;
+  public String getIconName(){
+    return iconName;
   }
 }

--- a/src/com/intellij/plugins/thrift/GeneratorIcons.java
+++ b/src/com/intellij/plugins/thrift/GeneratorIcons.java
@@ -7,7 +7,7 @@ import javax.swing.*;
 import java.util.HashMap;
 import java.util.Map;
 
-public class GeneratorIcons {
+public final class GeneratorIcons {
 
   public static Icon getIcon(GeneratorType generatorType) {
     return LazyHolder.INSTANCE.getIconInstance(generatorType);
@@ -24,13 +24,15 @@ public class GeneratorIcons {
   private Map<GeneratorType, IconLoader.LazyIcon> icons = new HashMap<GeneratorType, IconLoader.LazyIcon>();
 
   private GeneratorIcons() {
+
     for (final GeneratorType generatorType : GeneratorType.values()) {
-      icons.put(generatorType, new IconLoader.LazyIcon() {
+      IconLoader.LazyIcon lazyIcon = new IconLoader.LazyIcon() {
         @Override
         protected Icon compute() {
           return IconLoader.getIcon(generatorType.getIconName());
         }
-      });
+      };
+      icons.put(generatorType, lazyIcon);
     }
   }
 }

--- a/src/com/intellij/plugins/thrift/GeneratorIcons.java
+++ b/src/com/intellij/plugins/thrift/GeneratorIcons.java
@@ -1,0 +1,36 @@
+package com.intellij.plugins.thrift;
+
+import com.intellij.openapi.util.IconLoader;
+import com.intellij.plugins.thrift.config.target.GeneratorType;
+
+import javax.swing.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GeneratorIcons {
+
+  public static Icon getIcon(GeneratorType generatorType) {
+    return LazyHolder.INSTANCE.getIconInstance(generatorType);
+  }
+
+  private Icon getIconInstance(GeneratorType generatorType) {
+    return this.icons.get(generatorType);
+  }
+
+  private static class LazyHolder {
+    static final GeneratorIcons INSTANCE = new GeneratorIcons();
+  }
+
+  private Map<GeneratorType, IconLoader.LazyIcon> icons = new HashMap<GeneratorType, IconLoader.LazyIcon>();
+
+  private GeneratorIcons() {
+    for (final GeneratorType generatorType : GeneratorType.values()) {
+      icons.put(generatorType, new IconLoader.LazyIcon() {
+        @Override
+        protected Icon compute() {
+          return IconLoader.getIcon(generatorType.getIconName());
+        }
+      });
+    }
+  }
+}

--- a/src/com/intellij/plugins/thrift/config/facet/ThriftFacetConf.java
+++ b/src/com/intellij/plugins/thrift/config/facet/ThriftFacetConf.java
@@ -20,6 +20,7 @@ import com.intellij.openapi.util.WriteExternalException;
 import com.intellij.openapi.vfs.VfsUtil;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.openapi.vfs.VirtualFileManager;
+import com.intellij.plugins.thrift.GeneratorIcons;
 import com.intellij.plugins.thrift.ThriftBundle;
 import com.intellij.plugins.thrift.config.ThriftCompilerOptions;
 import com.intellij.plugins.thrift.config.facet.options.OptionsDialogWrapper;
@@ -238,7 +239,7 @@ public class ThriftFacetConf implements FacetConfiguration, PersistentStateCompo
             assert value != null;
 
             GeneratorType type = (GeneratorType)value;
-            setIcon(type.getIcon());
+            setIcon(GeneratorIcons.getIcon(type));
             append(type.name(), SimpleTextAttributes.REGULAR_ATTRIBUTES);
           }
         });
@@ -339,7 +340,7 @@ public class ThriftFacetConf implements FacetConfiguration, PersistentStateCompo
         private final GeneratorType myType;
 
         public CreateGeneratorAction(GeneratorType type) {
-          super(type.name(), "", type.getIcon());
+          super(type.name(), "", GeneratorIcons.getIcon(type));
           myType = type;
         }
 

